src/basic.ts

```diff
@@ -0,0 +1,27 @@
+/** Adapted from https://github.com/krzkaczor/ts-essentials */
+
+/** Essentials */
+export type Primitive = string | number | boolean | undefined | null;
+
+/** Dictionaries related */
+export type Dictionary<T, K extends string | number = string> = { [key in K]: T };
+export type DictionaryValues<T> = T extends Dictionary<infer U> ? U : never;
+
+/** Like Partial but recursive */
+export type DeepPartial<T> = {
```

Comment:
I do not see any usage of these around — I can live with that although I'm more on "do not put stuff you're not using"

Reply:
Being used (copied right now since we don't have a single place to put the definitions) in several places (Formtron and Core are two that I can think of off the top of my head).